### ISSUE TITLE
Problem with loss during training (nan)

### ISSUE THREAD

**Issue by johnny-mueller** [zzh8829/yolov3-tf2#249]
Hello, I tried to use the COCO record first to get the odell running first. This is where the problem has already arisen. If the model can be trained I want to preprocess the COCO data set and then train the actual model.

**Problem:**
When I train the model I always get a loss which is nan after a while (eager_tf mode):

```
I0430 10:29:54.313171  7496 train.py:149] 1_train_0, 14398.1630859375, [624.42706, 2601.958,11159.701]
I0430 10:29:57.046593  7496 train.py:149] 1_train_1, nan, [1835.8336, 3319.57, nan]
I0430 10:29:59.713561  7496 train.py:149] 1_train_2, nan, [nan, nan, nan]
I0430 10:30:02.475476  7496 train.py:149] 1_train_3, nan, [nan, nan, nan]
I0430 10:30:05.240971  7496 train.py:149] 1_train_4, nan, [nan, nan, nan]
```
I have also looked at some previous issues and tried to fix the problem but have not yet come to a conclusion.

**Previous Approach:**
1. I have downloaded and converted the weights. As far as I can see there were no problems. And when I call detect.py with the converted weights, the corresponding ouput is delivered.
2. To generate the Tfrecord files I used the script [create_coco_tf_record.py](https://github.com/tensorflow/models/blob/master/research/object_detection/dataset_tools/create_coco_tf_record.py).
I have created 100 files for the training set, 10 files for the validation set and 100 for testdev as defined in the script as default.
3. I used the generated tfrecord files directly to train the model, and the output shown above is the result.

**Exemplary content of the TFrecord file**
```
features {
  feature {
    key: "image/encoded"
    value {
      bytes_list {
        value: "\377\330\377\340\000\020JFIF\000\001\001\001\000H\000H\000\000\377\342\014XICC_PROFILE\000\001\001\000\ [Remaining part was removed for space reasons]"
      }
    }
  }
  feature {
    key: "image/filename"
    value {
      bytes_list {
        value: "000000391895.jpg"
      }
    }
  }
  feature {
    key: "image/format"
    value {
      bytes_list {
        value: "jpeg"
      }
    }
  }
  feature {
    key: "image/height"
    value {
      int64_list {
        value: 360
      }
    }
  }
  feature {
    key: "image/key/sha256"
    value {
      bytes_list {
        value: "d0d62a2eed6433c8d65bae1c8dca849eb678d35a8f415e4a11e5daada24eb93a"
      }
    }
  }
  feature {
    key: "image/object/area"
    value {
      float_list {
        value: 12190.4453125
        value: 14107.271484375
        value: 708.2605590820312
        value: 626.9852294921875
      }
    }
  }
  feature {
    key: "image/object/bbox/xmax"
    value {
      float_list {
        value: 0.7369062304496765
        value: 0.7714999914169312
        value: 0.7930625081062317
        value: 0.8072500228881836
      }
    }
  }
  feature {
    key: "image/object/bbox/xmin"
    value {
      float_list {
        value: 0.561203122138977
        value: 0.5310624837875366
        value: 0.7369375228881836
        value: 0.759390652179718
      }
    }
  }
  feature {
    key: "image/object/bbox/ymax"
    value {
      float_list {
        value: 0.9992777705192566
        value: 0.8969166874885559
        value: 0.6136666536331177
        value: 0.6063610911369324
      }
    }
  }
  feature {
    key: "image/object/bbox/ymin"
    value {
      float_list {
        value: 0.40602776408195496
        value: 0.06155555695295334
        value: 0.4800555408000946
        value: 0.5091944336891174
      }
    }
  }
  feature {
    key: "image/object/class/text"
    value {
      bytes_list {
        value: "motorcycle"
        value: "person"
        value: "person"
        value: "bicycle"
      }
    }
  }
  feature {
    key: "image/object/is_crowd"
    value {
      int64_list {
        value: 0
        value: 0
        value: 0
        value: 0
      }
    }
  }
  feature {
    key: "image/source_id"
    value {
      bytes_list {
        value: "391895"
      }
    }
  }
  feature {
    key: "image/width"
    value {
      int64_list {
        value: 640
      }
    }
  }
}
```

At the moment I have no further approach how to get the model to run and train. Is there one approach that I can try?

**Comment by Mysterion626** [comment]
Hi,
You have to edit the /data/coco.names file. The class names are misspeld for example:
"diningtable" but the correct class name is "dining table".
See: https://github.com/amikelive/coco-labels/blob/master/coco-labels-2014_2017.txt for corect names

**Comment by johnny-mueller** [comment]
@Mysterion626 Thank you very much for the hint. I had actually examined all the data and also the code. I had actually not placed the file under suspicion. Thank you. The model is already in training.
